pkg/core/object/replicate.go

```diff
@@ -30,18 +31,20 @@ const (
 //	"network": network magic
 //	"cid": _raw_ container ID (32 bytes)
 //	"oid": _raw_ object ID (32 bytes)
+//	"firstPart": _raw_ object ID (32 bytes)
 //	"size": payload size
 //	"deleted": array of _raw_ object IDs
 //	"locked": array of _raw_ object IDs
 //	"validuntil": last valid block number for meta information
 //
 // Last valid epoch is object's creation epoch + 10.
-func EncodeReplicationMetaInfo(cID cid.ID, oID oid.ID, pSize uint64,
+func EncodeReplicationMetaInfo(cID cid.ID, oID, firstPart oid.ID, pSize uint64,
 	deleted, locked []oid.ID, vub uint64, magicNumber uint32) []byte {
 	kvs := []stackitem.MapElement{
 		kv(networkMagicKey, magicNumber),
 		kv(cidKey, cID[:]),
 		kv(oidKey, oID[:]),
+		kv(firstPartKey, firstPart[:]),
```

Comment:
i dont suggest to push missing `firstPart` as zero ID, it's a reserved invalid value. Push emptiness instead: this will match the zero length condition from ur contract change https://github.com/nspcc-dev/neofs-contract/pull/457/commits/416936f4c6344c5745ae863e25fb0ee5c23e708c

Reply:
added optional fields support. `firstPart` is now one of them